src/Web/OIDC/Client/CodeFlow.hs

```diff
@@ -169,11 +170,15 @@ validateIdToken oidc jwt' = do
     let jwks = P.jwkSet . oidcProvider $ oidc
         token = Jwt.unJwt jwt'
         alg = fmap (Jwt.JwsEncoding . P.getJwsAlg)
-                . listToMaybe
                 . P.idTokenSigningAlgValuesSupported
                 . P.configuration
                 $ oidcProvider oidc
-    decoded <- Jwt.decode jwks alg token
+    decoded <-
+        (\x -> case partitionEithers x of
```

Comment:
Is this case match total?
Since we're already in a "do" block, would it be simpler if we split this into a few statements?
Both cases result in a "pure" call. Does that mean the bind can become fmap?

Reply:
Am I reading this right? You're taking the first success, otherwise you're taking the first failure? That seems sensible. There's an edge case where `x` is empty.

Reply:
Yeah Its not total and it probably should be, I'll rework it so it is.  You are correct, we only care about failure if there are no successes.